test/e2e/capi_test.go

```diff
@@ -56,7 +56,14 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 		Expect(os.Setenv(AzureVNetName, fmt.Sprintf("%s-vnet", rgName))).To(Succeed())
 
 		Expect(e2eConfig.Variables).To(HaveKey(capi_e2e.KubernetesVersionUpgradeFrom))
-		Expect(os.Setenv("WINDOWS_WORKER_MACHINE_COUNT", "2")).To(Succeed())
+
+		Expect(e2eConfig.Variables).To(HaveKey(capi_e2e.KubernetesVersionUpgradeFrom))
+		v, err := semver.ParseTolerant(e2eConfig.GetVariable(capi_e2e.KubernetesVersionUpgradeFrom))
+		Expect(err).NotTo(HaveOccurred())
+		// Opt into Windows for versions greater than or equal to 1.23
+		if v.GTE(semver.MustParse("1.23.0")) {
```

Comment:
@CecileRobertMichon wouldn't we want to use an entirely different cluster template rather than simply rely on this change to `$WINDOWS_WORKER_MACHINE_COUNT`?